### PR TITLE
Use spack_get_libs.sh to setup LD_LIBRARY_PATH

### PR DESCRIPTION
**Pull request by b10n1k**
Spack doesnt set LD_LIBRARY_PATH and libraries cant be found. Apply workaround
from bsc#1208751. This is supposed is gonna be fixed and this workaround would
be needed.

Update: Second commit overrides workaround. 

Signed-off-by: Ioannis Bonatakis ybonatakis@suse.com

https://progress.opensuse.org/issues/127481
http://aquarius.suse.cz/tests/17841 - w/o HPC_LIB
http://aquarius.suse.cz/tests/17835 - with HPC_LIB

### CODE REVIEW

**Comment by pevik** [comment]
From the first commit message:
> This is supposed is gonna be fixed and this workaround would be needed.

I don't understand to the sentense. Do you mean this? This is supposed is _to_ be fixed and this workaround would _not_ be needed.

File: lib/hpcbase.pm [hunk 1]
```diff
@@ -355,44 +355,43 @@ After C<prepare_spack_env> run, C<spack> should be ready to build entire tool st
...
+    zypper_call "in spack", timeout => 1200;
     type_string('pkill -u root');    # this kills sshd
     select_serial_terminal(0);
-    assert_script_run "module load gnu $mpi";
-    assert_script_run 'source /usr/share/spack/setup-env.sh';
-    if (check_var('HPC_LIB', 'boost')) {
-        # Workaround for bsc#1208751
-        assert_script_run "spack config add modules:prefix_inspections:lib64:[LD_LIBRARY_PATH]";
-        assert_script_run "spack config add modules:prefix_inspections:lib:[LD_LIBRARY_PATH]";
-    }
```
Comment (pevik) [inline lib/hpcbase.pm]: Am I correct, that you add this code in previous commit and then you remove it? Why?

File: tests/hpc/spack_master.pm [hunk 2]
```diff
@@ -41,7 +43,9 @@ sub run ($self) {
 
     barrier_wait('MPI_SETUP_READY');
     if (check_var('HPC_LIB', 'boost')) {
-        assert_script_run 'spack load boost';
+        assert_script_run 'source /usr/share/spack/setup-env.sh';
```
Comment (pevik) [inline tests/hpc/spack_master.pm:46]: I suppose we will always use `/bin/sh` as symlink to `bash`, but `. /usr/share/spack/setup-env.sh` would work also on `dash`.

File: tests/hpc/spack_slave.pm [hunk 3]
```diff
@@ -25,9 +25,13 @@ sub run ($self) {
     type_string("$testapi::password\n");
     barrier_wait('CLUSTER_PROVISIONED');
     barrier_wait('MPI_SETUP_READY');
-    assert_script_run "spack load $mpi";
-    script_run "module av";
-
+    if (check_var('HPC_LIB', 'boost')) {
+        assert_script_run 'source /usr/share/spack/setup-env.sh';
```
Comment (pevik) [inline tests/hpc/spack_slave.pm:29]: Also here.

**Comment by b10n1k** [comment]
> From the first commit message:
> 
> > This is supposed is gonna be fixed and this workaround would be needed.
> 
> I don't understand to the sentense. Do you mean this? This is supposed is _to_ be fixed and this workaround would _not_ be needed.

I tried to do it in two steps with two consequential commits. The first has the work around. Next is the way to go as it was provided by Egbert. The gh message gets the initial commit message. Sorry for the confusion.

File: lib/hpcbase.pm (hunk 1 above)
Comment (b10n1k) [reply lib/hpcbase.pm]: That was the solution provided initially. Later on HPC team released a script to do that. It is what the script should do which is to set LD_LIBRARY_PATH such as the library can be found

File: tests/hpc/spack_master.pm (hunk 2 above)
Comment (b10n1k) [reply tests/hpc/spack_master.pm:46]: AFAIK the first is true and we really dont use dash. lets leave it as it is.

File: tests/hpc/spack_slave.pm (hunk 3 above)
Comment (b10n1k) [reply tests/hpc/spack_slave.pm:29]: As per last reply too. lets keep it as it is if there is no problem

**Comment by b10n1k** [comment]
i am gonna merge it as discuss offline. i hope it will be fine

**Comment by pevik** [comment]
Sure, no problem.

**Pull request merged**